Assets/_Scripts/BattleController.cs

```diff
@@ -0,0 +1,217 @@
+                playerActionCounter = 0;
+                this.ExecutePlayerTurn();
+            }
+        }
+    }
+
+    private bool hasEveryoneEnteredActions() {
+        foreach (var player in _players) {
+            if (player.hasSetCommand == false) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+
+    public void UpdateBattleOptionUI() {
+        selectionCursor.transform.parent = battleOptionsUI[(int)(_heroPlayer.command.option)].transform;
+        selectionCursor.anchoredPosition = selectionCursorOffset;
+    }
+
+
+    public void ExecutePlayerTurn() {
+        this.startActions = true;
+        this.ActionMenu.gameObject.SetActive(false);
+        this.CommentaryMenu.gameObject.SetActive(true);
+
+        // Sort by player speed
+        List<FightingEntity> orderedPlayers = new List<FightingEntity>(GameManager.Instance.getAllFightingEntities());
+        orderedPlayers.Sort( (FightingEntity a, FightingEntity b) =>  {  return b.stats.GetSpeed().CompareTo(a.stats.GetSpeed()); });
+
+        this.DoActionHelper(orderedPlayers, 0);
+        
+    }
+
+
+    public void DoActionHelper(List<FightingEntity> orderedEntities, int curIndex) {
+        if (curIndex == orderedEntities.Count) {
+            this.OnPlayerTurnStart();
+            return;
+        }
+        
+        FightingEntity entity = orderedEntities[curIndex];
+        FightingEntity enemyEntity;
+        if (entity.GetType() == typeof(Enemy)) {
+            entity.command.target = getRandomEnemyTarget();
+            enemyEntity = _players[entity.command.target];
+        } else {
+            enemyEntity = GameManager.Instance.enemies[entity.command.target];
+        }
+
+        StartCoroutine(dummyAttackAnimation(entity, enemyEntity, orderedEntities, curIndex));
+        
+    }
+
+    public int getRandomEnemyTarget() {
+        int enemyTarget = Random.Range(0, _players.Count);
+        return enemyTarget;
+    }
+
+    IEnumerator dummyAttackAnimation(FightingEntity a, FightingEntity b, List<FightingEntity> orderedEntities, int index) {
+
+        string attackerName = a.Name;
+        string attackName = a.command.option.ToString(); // TODO: Better names
+
+        CommentaryText.text = "" + attackerName + " used " + attackName;
+        yield return new WaitForSeconds(2f);
+        this.DoAction(a, b);
+        this.DoActionHelper(orderedEntities, index + 1);
+    }
+
+    public void OnPlayerTurnStart() {
+        this.startActions = false;
+        
+        this.ActionMenu.gameObject.SetActive(true);
+        this.CommentaryMenu.gameObject.SetActive(false);
+
+        this.playerActionCounter = 0;
+        foreach (var player in _players) {
+            player.hasSetCommand = false;
+
+            if (player.modifiers.Contains("defend")) {
+                player.modifiers.Remove("defend");
+            }
+        }
+    }
+
+    public void DoAction(FightingEntity a, FightingEntity b) {
```

Comment:
I would suggest we actually make executing actions a class in the future. This allows us for slightly more complex involvement.